### PR TITLE
Move to fvm_shared::crypto

### LINKED ISSUE #1575: Remove `crypto` crate and use `fvm_shared::crypto` instead.
**Issue summary**

 - [ ] Update forest code base to use `fvm_shared::crypto::*` instead of `crypto::*`.
 - [ ] Delete `crypto` crate.

**Other information and links**

All of the `crypto` crate may not be available from `fvm_shared::crypto`. If something isn't available, consider if we need to keep it in our own crypto crate.

### PR DESCRIPTION
**Pull request by connormullett**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes 
#1575 

**Other information and links**

### CODE REVIEW

File: key_management/src/keystore.rs [hunk 1]
```diff
@@ -80,7 +80,7 @@ pub mod json {
     #[derive(Serialize, Deserialize)]
     struct JsonHelper {
         #[serde(rename = "Type")]
-        sig_type: SignatureTypeJson,
+        sig_type: SignatureType,
```
Comment (lemmih) [inline key_management/src/keystore.rs:83]: This appears to be a mistake.

**Comment by lemmih** [comment]
Try to keep the original module layout of the `forest_crypto` crate. That way, git will be able to see that you only removed code and didn't actually write a completely new JSON serializer.

Also, you accidentally restored the `vm/runtime` crate. You probably didn't mean to do that.

**Comment by connormullett** [comment]
> Try to keep the original module layout of the `forest_crypto` crate. That way, git will be able to see that you only removed code and didn't actually write a completely new JSON serializer.
> 
> Also, you accidentally restored the `vm/runtime` crate. You probably didn't mean to do that.

I think I got it all back to where it was now. Should be good to go

**Comment by lemmih** [comment]
The `price_list.rs` file was restored by accident, it seems.

**Comment by connormullett** [comment]
> The `price_list.rs` file was restored by accident, it seems.

:+1:

File: node/rpc/src/state_api.rs [hunk 2]
```diff
@@ -24,12 +24,12 @@ use blocks::{
 use blockstore::{BlockStore, BlockStoreExt};
 use bls_signatures::Serialize as SerializeBls;
 use cid::{json::CidJson, Cid, Code::Blake2b256};
-use crypto::SignatureType;
 use fil_types::{
     verifier::{FullVerifier, ProofVerifier},
     PoStProof,
 };
 use fvm_shared::bigint::BigInt;
+use fvm_shared::crypto::signature::SignatureType;
```
Comment (lemmih) [inline node/rpc/src/state_api.rs:32]:
```suggestion
use fvm_shared::crypto::signature::{Signature, SignatureType};
```

File: node/rpc/src/state_api.rs [hunk 3]
```diff
@@ -655,9 +655,9 @@ pub(crate) async fn miner_create_block<
     )?;
 
     let calculated_bls_agg = if bls_sigs.is_empty() {
-        Some(crypto::Signature::new_bls(vec![]))
+        Some(fvm_shared::crypto::signature::Signature::new_bls(vec![]))
```
Comment (lemmih) [inline node/rpc/src/state_api.rs:658]:
```suggestion
        Some(Signature::new_bls(vec![]))
```

File: node/rpc/src/state_api.rs [hunk 4]
```diff
@@ -655,9 +655,9 @@ pub(crate) async fn miner_create_block<
...
     } else {
-        Some(crypto::Signature::new_bls(
+        Some(fvm_shared::crypto::signature::Signature::new_bls(
```
Comment (lemmih) [inline node/rpc/src/state_api.rs:660]:
```suggestion
        Some(Signature::new_bls(
```

File: tests/serialization_tests/tests/signing_ser.rs [hunk 5]
```diff
@@ -6,10 +6,11 @@
 
 use bls_signatures::{PrivateKey, Serialize};
 use cid::Cid;
-use crypto::{signature, Signature};
 use encoding::Cbor;
+use forest_crypto;
```
Comment (lemmih) [inline tests/serialization_tests/tests/signing_ser.rs:10]:
```suggestion
use forest_crypto::signature;
```

File: tests/serialization_tests/tests/signing_ser.rs [hunk 6]
```diff
@@ -23,7 +24,7 @@ struct TestVec {
     unsigned: Message,
     cid: String,
     private_key: String,
-    #[serde(with = "signature::json")]
+    #[serde(with = "forest_crypto::signature::json")]
```
Comment (lemmih) [inline tests/serialization_tests/tests/signing_ser.rs:27]:
```suggestion
    #[serde(with = "signature::json")]
```

File: vm/message/src/signed_message.rs [hunk 7]
```diff
@@ -127,7 +130,6 @@ pub mod json {
     use super::*;
 
     use cid::Cid;
-    use crypto::signature;
```
Comment (lemmih) [inline vm/message/src/signed_message.rs:130]: Maybe change this to `use forest_crypto::signature`

File: vm/message/src/signed_message.rs [hunk 8]
```diff
@@ -161,7 +163,7 @@ pub mod json {
         struct SignedMessageSer<'a> {
             #[serde(with = "message::json")]
             message: &'a Message,
-            #[serde(with = "signature::json")]
+            #[serde(with = "forest_crypto::signature::json")]
```
Comment (lemmih) [inline vm/message/src/signed_message.rs:166]:
```suggestion
            #[serde(with = "signature::json")]
```

File: vm/message/src/signed_message.rs [hunk 9]
```diff
@@ -183,7 +185,7 @@ pub mod json {
         struct SignedMessageDe {
             #[serde(with = "message::json")]
             message: Message,
-            #[serde(with = "signature::json")]
+            #[serde(with = "forest_crypto::signature::json")]
```
Comment (lemmih) [inline vm/message/src/signed_message.rs:188]:
```suggestion
            #[serde(with = "signature::json")]
```

**Review by lemmih — CHANGES_REQUESTED** [review]
Good. Very close to being done. Could you investigate whether we can remove the `blst` and `pairing` features for `forest_crypto`?

File: Makefile [hunk 10]
```diff
@@ -75,17 +75,17 @@ test-vectors: pull-serialization-tests run-vectors
 test:
 	cargo test --all --exclude serialization_tests --exclude forest_message --exclude forest_crypto -- --test-threads=$(RUST_TEST_THREADS)
 	cargo test -p forest_crypto --features blst --no-default-features -- --test-threads=$(RUST_TEST_THREADS)
+	cargo test --all --exclude serialization_tests --exclude forest_message -- --test-threads=$(RUST_TEST_THREADS)
 	# FIXME: https://github.com/ChainSafe/forest/issues/1444
```
Comment (LesnyRumcajs) [inline Makefile:79]: should this comment go as well in this case?

File: Makefile [hunk 11]
```diff
@@ -75,17 +75,17 @@ test-vectors: pull-serialization-tests run-vectors
...
-	#cargo test -p forest_crypto --features pairing --no-default-features -- --test-threads=$(RUST_TEST_THREADS)
```
[the remark this answers is not in the document]
Comment (connormullett) [reply Makefile:79]: looks like this was an error when i restored it back to what it was originally. When I removed the crate initially, I cut out the related tests and messed up putting them back in. Good catch

File: Makefile (hunk 10 above)
Comment (connormullett) [reply Makefile:79]: Yea I messed up restoring the rule here, It should have been

File: node/rpc/Cargo.toml [hunk 12]
```diff
@@ -60,6 +56,8 @@ state_tree = { path = "../../vm/state_tree" }
 wallet = { package = "key_management", path = "../../key_management", features = [
     "json",
 ] }
+forest_crypto = { version = "0.5.2", features = ["blst", "json"] }
+fil_actors_runtime = "=7.5.1"
```
Comment (lemmih) [inline node/rpc/Cargo.toml:60]: Like before, this should be deleted or have a different version number.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: blockchain/chain_sync/Cargo.toml [hunk 13]
```diff
@@ -52,6 +51,8 @@ time = { version = "0.3", features = ["serde"] }
 prometheus = { version = "0.12.0", features = ["process"] }
 lazy_static = "1.4.0"
 statediff = { path = "../../utils/statediff", optional = true }
+forest_crypto = { version = "0.5.2", features = ["blst"] }
+fil_actors_runtime = "=7.5.1"
```
[the remark this answers is not in the document]
Comment (connormullett) [reply blockchain/chain_sync/Cargo.toml]: :+1:

File: node/rpc/Cargo.toml (hunk 12 above)
Comment (connormullett) [reply node/rpc/Cargo.toml]: Should be good now

**Review by tyshko5 — APPROVED** [review]

**Pull request merged**